Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -99,7 +102,18 @@ class MessageQueueManager {
         }
     }
 
-    private func handleMessage(message: Message) {
+    func processFetchResponse(_ fetchedMessages: [Message]) {
+        // To prevent us from showing expired / revoked messages, reset the local queue with the latest queue from the backend service.
+        // The backend service is the single-source-of-truth for in-app messages for each user.
+        clearUserMessagesFromLocalStore()
+        addMessagesToLocalStore(messages: fetchedMessages)
```

Comment:
I thought of combining them but decided not to because these functions are used in multiple places in the code besides here. I didn't want to modify these functions to add side effects to the functions. 

I would be OK with creating a new function `replace` that internally calls these 2 functions.